### PR TITLE
Fix-6080: Update the CNI version to 1.0.0

### LINKED ISSUE #6080: contrib/cni/README.md suggests installing outdated CNI version 0.3.1, not supported by Kubernetes
### What happened?

According to https://kubernetes.io/docs/concepts/extend-kubernetes/compute-storage-net/network-plugins/ Kubernetes supports CNI-plugins with minimum specification version 0.4.0, but prefers version 1.0.0.

https://github.com/cri-o/cri-o/tree/main/contrib/cni#configuration-directory suggests copying files from `cri-o.git/contrib/cni` to `/etc/cni/net.d`.  The files in  `cri-o.git/contrib/cni` contain `"cniVersion": "0.3.1"`.

Please elaborate why it is suggested to utilize the 0.3.1 interface, instead of a newer interface, supported by Kubernetes?

### What did you expect to happen?

Not applicable.

### How can we reproduce it (as minimally and precisely as possible)?

Not applicable.

### Anything else we need to know?

Not applicable.

### CRI-O and Kubernetes version

Not applicable.

### OS version

Not applicable.

### Additional environment details (AWS, VirtualBox, physical, etc.)

Not applicable.

### PR DESCRIPTION
**Pull request by amarlearning**
#### What type of PR is this?

/kind bug

#### What this PR does / why we need it:

Updated the CNI version of the network config files present in `contrib/cni/`.

#### Which issue(s) this PR fixes:
Fixes #6080 

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Review by saschagrunert — APPROVED** [review]
/ok-to-test

**Comment by dilyanpalauzov** [comment]
My understanding is that CNI configuration files need the .conflist extension, if they have `"plugins":[]` element.

For a file with plugins element, containing "cniVersion": "1.0.0", "plugins": [...] with .conf extension CRI-O 1.25.1 reports:

Oct 11 09:22:48 d crio[55802]: time="2022-10-11 09:22:48.377791412+03:00" level=error msg="Error loading CNI config file /etc/cni/net.d/11-crio-ipv4-bridge.conf: error parsing configuration: missing 'type'"

At the same time CRI-O 1.25.1 has no problems to load:

```json
{
    "cniVersion": "1.0.0",
    "type": "loopback"
}
```
My question is: isn’t it sufficient just to change the `"cniVersion"` lines?

At the same time the CNI Specification 0.3.1 at https://github.com/containernetworking/cni/blob/spec-v0.3.1/SPEC.md distinguishes between:
* Network Configuration — https://github.com/containernetworking/cni/blob/spec-v0.3.1/SPEC.md#network-configuration , which does not contain `"plugins":`, and
* Network Configuration List — https://github.com/containernetworking/cni/blob/spec-v0.3.1/SPEC.md#network-configuration-lists, which contains `"plugins":`

While the 1.0.0 specification at https://github.com/containernetworking/cni/blob/spec-v1.0.0/SPEC.md does not contain plugin configuration without `"plugins"` element.  But the examples at https://github.com/containernetworking/cni/blob/spec-v1.0.0/SPEC.md#appendix-examples , which are not files but json files exchanged on the wire, contain JSON snippets with `"type"` without `"plugins"`.

**Comment by amarlearning** [comment]
@dilyanpalauzov The network configuration format for the CNI specification v1.0.0 defines the format [here](https://github.com/containernetworking/cni/blob/spec-v1.0.0/SPEC.md#configuration-format)

**Comment by haircommander** [comment]
> 

I think we do need the config for the loopback plugin though, we just don't use an external binary to do that configuration

**Comment by dilyanpalauzov** [comment]
As I mentioned here earlier, on my system CRI-O 1.25.1 loads files from /etc/cni/net.dev/ with `plugins:[…]` element, only if the files have `.conflist` extension.

**Comment by haircommander** [comment]
@amarlearning can you rebase to update merge conflict and grab test fixes please?

File: contrib/bundle/build [hunk 1]
```diff
@@ -36,7 +36,7 @@ FILES_ETC=(
     "$GIT_ROOT/contrib/bundle/10-crun.conf"
 )
 FILES_CONTRIB=(
-    "$GIT_ROOT/contrib/cni/10-crio-bridge.conf"
+    "$GIT_ROOT/contrib/cni/10-crio-bridge.conflist"
```
Comment (haircommander) [inline contrib/bundle/build:39]: why the rename to conflist?

Comment (TomSweeneyRedHat) [reply contrib/bundle/build:39]: That's my question too.  If the name change is valid, the rest LGTM.

Comment (dilyanpalauzov) [reply contrib/bundle/build:39]: My experiments show, that when the configuration file has `plugins:` then it must have also .conflist extension.

File: tutorials/kubeadm.md [hunk 2]
```diff
@@ -9,7 +9,7 @@ kubeadm expects a POD Network CIDR (`--pod-network-cidr`) to be defined when you
 
 | CNI Plugin              | CIDR          | Notes                                                                                                                                         |
 | ----------------------- | ------------- | --------------                                                                                                                                |
-| Bridge plugin (default) | 10.85.0.0/16  | The default bridge plugin is defined [here](/contrib/cni/10-crio-bridge.conf). This is only suitable when your cluster has a **single node**. |
+| Bridge plugin (default) | 10.85.0.0/16  | The default bridge plugin is defined [here](/contrib/cni/10-crio-bridge.conflist). This is only suitable when your cluster has a **single node**. |
 | Flannel                 | 10.224.0.0/16 | This is a good choice for clusters with multiple nodes.                                                                                       |
```
Comment (dilyanpalauzov) [inline tutorials/kubeadm.md:13]: The table is not anymore aligned.

File: contrib/bundle/build (hunk 1 above)
Comment (dilyanpalauzov) [reply contrib/bundle/build:39]: See also https://github.com/cri-o/cri-o/pull/6289#issuecomment-1274156122.

Comment (amarlearning) [reply contrib/bundle/build:39]: @haircommander As @dilyanpalauzov mentioned the configurations file now has `plugins` so I have to change the extensions of the filename from `.conf` to `.conflist`.

Hence, this change.

**Comment by haircommander** [comment]
```
E1216 07:14:00.323917   43628 remote_runtime.go:222] "RunPodSandbox from runtime service failed" err="rpc error: code = Unknown desc = failed to get network status for pod sandbox k8s_create-PodSandbox-for-container-36897865-7d11-11ed-8b03-42010a000002_cri-test-namespace36897875-7d11-11ed-8b03-42010a000002_cri-test-uid36897871-7d11-11ed-8b03-42010a000002_2(84860bfe684a76e556d848df1e7923395f2898741867b1e9392d152de8230382): error checking pod cri-test-namespace36897875-7d11-11ed-8b03-42010a000002_create-PodSandbox-for-container-36897865-7d11-11ed-8b03-42010a000002 for CNI network \"crio\": Expected Route {Dst:{IP:1100:200::1 Mask:ffffff00000000000000000000000000} GW:<nil>} not found in routing table"
```
 seems to be the [issue](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/cri-o_cri-o/6289/pull-ci-cri-o-cri-o-main-ci-fedora-critest/1603644107899539456/artifacts/fedora-critest/cri-o-fedora-critest-test/build-log.txt)

**Comment by haircommander** [comment]
looks like this was added in [this](https://github.com/containernetworking/plugins/commit/74a259657318d29a5f3eb5ff8b9c1c7b74409be3) commit around 0.8.0 version of the plugins.

**Comment by haircommander** [comment]
@dcbw @aojea @squeed @mccv1r0 sorry for the wide reach, I am not sure who to ask here. upgrading to cni v1 plugins breaks our CI setup, and it's not clear to me if the box should be configured differently or the config. Is anyone around in the next day or so to advise/able to redirect to someone who can? I'm trying to get 1.26.0 out soon and this would be good to have.

File: contrib/cni/10-crio-bridge.conflist [hunk 3]
```diff
@@ -16,5 +18,7 @@
             [{ "subnet": "10.85.0.0/16" }],
             [{ "subnet": "1100:200::/24" }]
```
Comment (mccv1r0) [inline contrib/cni/10-crio-bridge.conflist:19]: Add a gateway for  the route for the `1100:200::/24"`  range; omit the host, `::1` in the prefix too.  You probably want `::/64` and not `::/24 `as well.

```{ "dst": "1100:200::/64", "gw": "fe80::1" }```  or if there really is a router with addr `1100:200::1`
```{ "dst": "1100:200::/64", "gw": "1100:200::1" }```

Comment (aojea) [reply contrib/cni/10-crio-bridge.conflist:19]: it is weird, the error comes from https://github.com/containernetworking/plugins/blob/ec76e3c35c55f3107c45a54ce310c9c7e98bac75/plugins/main/bridge/bridge.go#L961 , why it worked before?

Comment (mccv1r0) [reply contrib/cni/10-crio-bridge.conflist:19]: Perhaps the intention was for IPv6 default route ```::/0``` ?     Similar to how ```0.0.0.0/0``` is used for IPv4 

Then omitting gwy should be ok (this bridge `cni0` would be used as the gateway for the default IPv6 route.)

Comment (aojea) [reply contrib/cni/10-crio-bridge.conflist:19]: yeah, I think that if you set `IsGateway` to true you have to omit the default routes

Comment (amarlearning) [reply contrib/cni/10-crio-bridge.conflist:19]: Shall I add a route for the gateway or just set `IsGateway` to `false`?

Comment (mccv1r0) [reply contrib/cni/10-crio-bridge.conflist:19]: What was already there didn't/couldn't work.

Like v4, you need to use `default` destination.

```
                        "routes": [{
                                "dst": "0.0.0.0/0",
                                "dst": "::/0"
                        }],
```
works for me in cnitool, but 

```
        "routes": [
            { "dst": "0.0.0.0/0" },
            { "dst": "::/0" }
        ],
```

should as well

This plugin creates interfaces inside containers that point to this bridge `cni0` to communicate with other containers on this same bridge.  `routes` is used to add routes to the containers routing table, supplying `dst` to route to and `gwy `to reach that dst.  (`gwy` doesn't have to be this bridge, that's why it's required.) 

Using default for a route and `isGateway` is a shortcut that has the plugin create an IP addr for this bridge (  ` "bridge": "cni0"`) itself and use that as the gwy.

Comment (amarlearning) [reply contrib/cni/10-crio-bridge.conflist:19]: That makes sense, thanks for sharing your insight it helps.

File: contrib/bundle/install [hunk 4]
```diff
@@ -1,7 +1,7 @@
 set -x
 install $SELINUX -d -m 755 "$DESTDIR$CNIDIR"
 install $SELINUX -D -m 755 -t "$DESTDIR$OPT_CNI_BIN_DIR" cni-plugins/*
-install $SELINUX -D -m 644 -t "$DESTDIR$CNIDIR" contrib/10-crio-bridge.conf
+install $SELINUX -D -m 644 -t "$DESTDIR$CNIDIR" contrib/10-crio-bridge.conflist
```
Comment (haircommander) [inline contrib/bundle/install:4]: weirdly enough the path here was wrong. it should be contrib/cni/10-crio-bridge.conflist . I'm not sure why this worked before

Comment (haircommander) [reply contrib/bundle/install:4]: it's causing [this](https://github.com/cri-o/cri-o/actions/runs/3751631905/jobs/6373420977) error

Comment (amarlearning) [reply contrib/bundle/install:4]: Ya even I saw this error in the build now. Not sue how it was working up until now 😢

**Comment by haircommander** [comment]
/override ci/prow/ci-rhel-integration
/override ci/kata-jenkins
/lgtm

thanks for carrying this one through @amarlearning, I really appreciate it! and thanks @mccv1r0 and @aojea for your council. I'll fix up the remaining test issues

File: contrib/bundle/install (hunk 4 above)
Comment (AkihiroSuda) [reply contrib/bundle/install]: Is this issue planned to be fixed? 🙏 

Hitting this in the cri-tools CI
https://github.com/kubernetes-sigs/cri-tools/pull/1053#issuecomment-1368555519

Comment (amarlearning) [reply contrib/bundle/install]: I want to take this one.

@haircommander can you confirm whether you have started working on this or not?

Comment (amarlearning) [reply contrib/bundle/install]: Made a temp fix on [this](https://github.com/cri-o/cri-o/pull/6476) PR for this issue.

cc @haircommander @AkihiroSuda

Comment (haircommander) [reply contrib/bundle/install]: thanks for taking it on @amarlearning !

**Pull request merged**